crates/bevy_ecs/src/schedule/stage.rs

```diff
@@ -392,33 +334,34 @@ impl SystemStage {
             })
             .collect();
 
-        for index in self.uninitialized_at_start.drain(..) {
-            let container = &mut self.exclusive_at_start[index];
-            if let Some(index) = container.run_criteria() {
-                container.set_run_criteria(new_indices[index]);
-            }
-            container.system_mut().initialize(world);
-        }
-        for index in self.uninitialized_before_commands.drain(..) {
-            let container = &mut self.exclusive_before_commands[index];
-            if let Some(index) = container.run_criteria() {
-                container.set_run_criteria(new_indices[index]);
-            }
+        for (mut container, system_type) in self.uninitialized_systems.drain(..) {
             container.system_mut().initialize(world);
-        }
-        for index in self.uninitialized_at_end.drain(..) {
-            let container = &mut self.exclusive_at_end[index];
+
             if let Some(index) = container.run_criteria() {
                 container.set_run_criteria(new_indices[index]);
             }
-            container.system_mut().initialize(world);
-        }
-        for index in self.uninitialized_parallel.drain(..) {
-            let container = &mut self.parallel[index];
-            if let Some(index) = container.run_criteria() {
-                container.set_run_criteria(new_indices[index]);
+
+            match system_type {
+                SystemType::Parallel => {
+                    if container.system().is_exclusive() {
+                        // we don't know if exclusive until params are initialized (unless told)
+                        self.exclusive_at_start.push(container);
```

Comment:
Should we be bother deprecating `.exclusive_system()` if using the `SystemParam` won't allow using the insertion_point?

Reply:
> Joy: you can still specify the insertion point (.at_start(), .before_commands(), and .at_end() will work for any system), the line you highlighted is just to catch systems that didn't

Ah, yeah. I editted a different comment and should have probably deleted and added this comment to where the deprecation for `.exclusive_system()` is. I feel like we should probably remove the deprecation as it might be confusing for people who are coercing an exclusive system and using the insertion point configuration, because we don't have a full replacement for that usage.

Reply:
Nah, it can be deprecated. `.at_start()`, `.at_end()`, and `.before_commands()` will both coerce systems into being exclusive and configure the insertion point.

The section you highlighted exists to catch any `&mut World` systems whose insertion point wasn't specified.